lib/validator/task/mod.rs

```diff
@@ -1155,15 +1155,446 @@ where
         }
     };
 
+    let deserialization_duration = start.elapsed() - fetch_duration;
+
+    tracing::debug!(
+        "Fetched ({:?}) and deserialized ({:?}) batch of {} block(s) with {} total transactions",
+        fetch_duration,
+        deserialization_duration,
+        blocks.len(),
+        blocks.iter().map(|block| block.txdata.len()).sum::<usize>(),
+    );
+
     Ok(blocks)
 }
 
+fn handle_block_batch(
+    dbs: &Dbs,
+    blocks: &[Block],
+    event_tx: &Sender<Event>,
+) -> Result<(), error::Sync> {
+    let start = Instant::now();
+
+    let mut total_txs = 0;
+    let mut total_bmm_commitments = 0;
+    let mut total_events = 0;
+
+    // Do a single DB transaction for the entire batch. DB commits are a big part
+    // of the sync time, so we want to reduce the number of times we do this.
+    let mut rwtxn = dbs.write_txn()?;
```

Comment:
Can you take the `rwtxn` as an argument? It would be very easy to call this while a write txn already exists, it would be better for the caller to handle creating a `rwtxn`. Up to you if you want to commit in this function or let the caller commit the rwtxn.